### PR TITLE
WW-5334 Misc VelocityManager code cleanup

### PR DESCRIPTION
**Pull request by kusalk**
WW-5334
--

### CODE REVIEW

File: plugins/velocity/src/main/java/org/apache/struts2/views/velocity/VelocityManager.java [hunk 1]
```diff
@@ -488,62 +468,18 @@ private void applyDefaultConfiguration(ServletContext context, Properties proper
...
         }
-
         properties.setProperty("userdirective", userdirective);
     }
 
-    private void addDirective(StringBuilder sb, Class<?> clazz) {
-        sb.append(clazz.getName()).append(",");
-    }
-
-    private static String replace(String string, String oldString, String newString) {
```
Comment (kusalk) [inline plugins/velocity/src/main/java/org/apache/struts2/views/velocity/VelocityManager.java:516]: This must be ancient code to be implementing String#replace from scratch

**Comment by kusalk** [comment]
Hmm we need to rethink the Maven module setup to avoid cyclic dependencies

**Comment by lukaszlenart** [comment]
> Hmm we need to rethink the Maven module setup to avoid cyclic dependencies. I'll create a separate PR addressing this.

What do you mean by that? I was planning to drop support for Portlets in Struts 7.x

**Comment by kusalk** [comment]
> What do you mean by that?

Since `struts2-junit-plugin` depends on other plugins (such as `struts2-convention-plugin`, `struts2-portlet-plugin`, `struts2-velocity-plugin`), it's impossible to use `struts2-junit-plugin` within those plugins to write unit tests (it creates a cyclic dependency). We either need to move the unit tests for these affected plugins into their own modules (quite odd), or what would be better is to not have `struts2-junit-plugin` depend on any other plugins.

> I was planning to drop support for Portlets in Struts 7.x

That's fine, I'll try reorganise the modules slightly until then - otherwise I can't unit test `struts2-velocity-plugin`!

**Comment by kusalk** [comment]
Will rebase and reopen after review and merge of #732

File: plugins/velocity/src/test/java/org/apache/struts2/views/velocity/result/VelocityResultTest.java [hunk 2]
```diff
@@ -18,93 +18,100 @@
...
-import com.mockobjects.dynamic.Mock;
-import com.opensymphony.xwork2.ActionContext;
-import com.opensymphony.xwork2.ActionInvocation;
-import com.opensymphony.xwork2.ActionProxy;
-import com.opensymphony.xwork2.util.ValueStack;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.when;
 
+public class VelocityResultTest extends XWorkJUnit4TestCase {
```
Comment (kusalk) [inline plugins/velocity/src/test/java/org/apache/struts2/views/velocity/result/VelocityResultTest.java:42]: Converted this to JUnit4 and Mockito

File: plugins/velocity/src/test/java/org/apache/struts2/views/velocity/VelocityManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.util.Properties;
+
+import static org.apache.struts2.views.velocity.VelocityManager.KEY_VELOCITY_STRUTS_CONTEXT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThrows;
+
+public class VelocityManagerTest extends StrutsJUnit4TestCase {
```
Comment (kusalk) [inline plugins/velocity/src/test/java/org/apache/struts2/views/velocity/VelocityManagerTest.java:42]: Doesn't cover everything but it's a start :)

**Review by lukaszlenart — APPROVED** [review]
LGTM 👍

**Pull request merged**